### PR TITLE
Fix Breaking Build

### PR DESCRIPTION
**Pull request by ltegman**
Fixes issue with missing config.json breaking travis as well as updating our lint config, fixing lint issues, and fixing a broken test.

### CODE REVIEW

File: test/Bonfires.spec.js [hunk 1]
```diff
@@ -91,14 +91,14 @@ test('Bonfires test', t => {
     st.end();
   });
 
-  t.test('should find raw wiki hints from KBase', st =>{
+  t.test('should find raw wiki hints from KBase', st => {
     st.plan(1);
```
Comment (abhisekp) [inline test/Bonfires.spec.js]: remove all `plans`
just `end` is fine and better. :+

Comment (ltegman) [inline test/Bonfires.spec.js]: That is not within the scope of this change and not needed.

File: lib/utils/Utils.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ const Utils = {
   },
 
   clog: function(where, msg, obj) {
-    if (process.env.SERVER_ENV || this.logLevel < LOG_LEVEL_INFO) {
+    if (this.logLevel < LOG_LEVEL_INFO) {
```
Comment (abhisekp) [inline lib/utils/Utils.js]: I think this should be `process.env.NODE_ENV === 'production'` or something.

We don't want the log outputs on production. It's a mess. Will lower the response time heavily.

**Comment by abhisekp** [comment]
@ltegman i get an error while using `about` command

**Comment by ltegman** [comment]
Could you post the full error you're seeing?

**Comment by abhisekp** [comment]
@ltegman 
```text
GBot>  in| about @abhisekp 
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:40:17)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:41:17)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
    at thanksCommands.showInfoCallback (/media/Xtras/GitRepos/camperbot/lib/bot/cmds/thanks.js:114:13)
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:44:9)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)

```

**Comment by ltegman** [comment]
@abhisekp I looked into this and I think it's just an issue with the `apiServer` value in the example config being missing the leading `www`. I've added a commit to fix this. Let me know if making that change in your config fixes the issue for you.

**Comment by abhisekp** [comment]
@ltegman but i see the problem persists
```text
GBot>  in| about @abhisekp
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:40:17)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:41:17)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
    at thanksCommands.showInfoCallback (/media/Xtras/GitRepos/camperbot/lib/bot/cmds/thanks.js:114:13)
    at IncomingMessage.response.on.err (/media/Xtras/GitRepos/camperbot/lib/utils/HttpWrap.js:44:9)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:926:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by abhisekp** [comment]
@ltegman yep! That fixed it. :+1:

**Comment by raisedadead** [comment]
Thanks for the PR @ltegman I cant seem to invoke the bot, messed up my config somehow. @abhisekp has tested it locally.

I went through the changes and they look solid to me.

**Pull request merged**